//    Copyright 2022, farm-ng inc.
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.

syntax = "proto3";

package farm_ng.core.proto;

// By convention we use the following naming scheme for messages which are
// represetned by generic types in other languages (e.g. templates in C++).

// MesageNameG0ParameterName - e.g. OptionalG0Double which is
// std::optional<double> in c++

message OptionalG0Float {
  float value = 1;
  bool has_value = 2;
}

message OptionalG0Double {
  double value = 1;
  bool has_value = 2;
}

message OptionalG0Int32 {
  int32 value = 1;
  bool has_value = 2;
}

message OptionalG0Int64 {
  int64 value = 1;
  bool has_value = 2;
}

message OptionalG0Bool {
  bool value = 1;
  bool has_value = 2;
}

message OptionalG0String {
  string value = 1;
  bool has_value = 2;
}


message FileSystemPath { string path_string = 1; }
